src/nvim/ui.c

```diff
@@ -441,8 +441,12 @@ void ui_puts(uint8_t *str)
 
     if (p_wd) {  // 'writedelay': flush & delay each time.
       ui_flush();
-      uint64_t wd = (uint64_t)labs(p_wd);
-      os_delay(wd, false);
+      if (!((got_int || (did_emsg && force_abort) || current_exception))){
+        uint64_t wd = (uint64_t)labs(p_wd);
+        os_delay(wd, false);
+      } else {
+        p_wd = 0;
```

Comment:
what happens if you remove this? I think we would prefer not to modify 'writedelay' on ctrl-c, that would be a different kind of annoying ...

Reply:
You then have to `Ctrl-C` multiple times, e.g. after `C-w s`,which I think is good, and similar to how Vim behaves.

Reply:
what if a timer or something causes frequent redraws? At least a second `Ctrl-C` in a short while should reset `p_wd` if not the first. One can have keybindings that quickly sets `'wd'` to a sensible positive and negative value respectively.

Reply:
Yes, I agree that a second `ctrl-c` in a short while should set it to 0.
If this is too complicated, then setting it on the first instance should do it probably (like in this PR), since not doing so makes it harder to abort due to the loop via the "Type  :quit<Enter>  to exit Nvim" message (see my last comment).

```diff
diff --git i/src/nvim/normal.c w/src/nvim/normal.c
index e4310de5d..385bfde57 100644
--- i/src/nvim/normal.c
+++ w/src/nvim/normal.c
@@ -1186,6 +1186,10 @@ static void normal_check_interrupt(NormalState *s)
       }
       got_int = false;
     }
+    // Reset "writedelay" on second interrupt.
+    if (s->previous_got_int) {
+      p_wd = 0;
+    }
     s->previous_got_int = true;
   } else {
     s->previous_got_int = false;
```